cachi2/core/package_managers/npm.py

```diff
@@ -173,6 +185,189 @@ def get_sbom_components(self) -> list[dict[str, str]]:
         packages = self._dependencies if self.lockfile_version == 1 else self._packages
         return [{"name": package.name, "version": package.version} for package in packages]
 
+    def get_dependencies_to_download(self) -> Dict[str, Dict[str, Optional[str]]]:
+        """Return a Dict of URL dependencies to download."""
+        packages = self._dependencies if self.lockfile_version == 1 else self._packages
+        return {
+            package.resolved_url: {
+                "version": package.version,
+                "name": package.name,
+                "integrity": package.integrity,
+            }
+            for package in packages
+            if "file:" not in package.resolved_url
+        }
+
+
+def _update_vcs_url_with_full_hostname(vcs: str) -> str:
+    """Update VCS URL with full hostname.
+
+    Transform github:kevva/is-positive#97edff6
+    into git+ssh://github.com/kevva/is-positive.git#97edff6
+
+    :param vcs: VCS URL to be modified with full hostname and file extension
+    :return: Updated VCS URL
+    """
+    host, _, path = vcs.partition(":")
```

Comment:
It's probably easier to use `vsc.split(":")` here if you're not going to use the separator.

Reply:
I believe the main benefit of partition here is that it will divide the string only into two substrings based on the first occurrence of the separator. I don't expect more `:` in valid URL to show up but it suits me here better.